components/brave_federated_learning/brave_operational_profiling.cc

```diff
@@ -0,0 +1,201 @@
+  simulate_local_training_step_timer_.reset();
+  collection_slot_periodic_timer_.reset();
+}
+
+void BraveOperationalProfiling::LoadPrefs() {
+  last_checked_slot_ = local_state_->GetInteger(kLastCheckedSlotPrefName);
+  collection_id_ = local_state_->GetString(kCollectionIdPrefName);
+  collection_id_expiration_time_ =
+      local_state_->GetTime(kCollectionIdExpirationPrefName);
+}
+
+void BraveOperationalProfiling::SavePrefs() {
+  local_state_->SetInteger(kLastCheckedSlotPrefName, last_checked_slot_);
+  local_state_->SetString(kCollectionIdPrefName, collection_id_);
+  local_state_->SetTime(kCollectionIdExpirationPrefName,
+                        collection_id_expiration_time_);
+}
+
+void BraveOperationalProfiling::OnCollectionSlotStartTimerFired() {
+  simulate_local_training_step_timer_->Reset();
+}
+
+void BraveOperationalProfiling::OnSimulateLocalTrainingStepTimerFired() {
+  SendCollectionSlot();
+}
+
+void BraveOperationalProfiling::SendCollectionSlot() {
+  current_collected_slot_ = GetCurrentCollectionSlot();
+  if (current_collected_slot_ == last_checked_slot_) {
+    return;
+  }
+
+  MaybeResetCollectionId();
+
+  auto resource_request = std::make_unique<network::ResourceRequest>();
+  resource_request->url = GURL(federatedLearningUrl);
+  resource_request->headers.SetHeader("X-Brave-FL-Operational-Profile", "?1");
+
+  resource_request->credentials_mode = network::mojom::CredentialsMode::kOmit;
+  resource_request->method = "POST";
+
+  url_loader_ = network::SimpleURLLoader::Create(
+      std::move(resource_request), GetNetworkTrafficAnnotationTag());
+  url_loader_->AttachStringForUpload(BuildPayload(), "application/json");
+
+  url_loader_->DownloadHeadersOnly(
+      url_loader_factory_.get(),
+      base::BindOnce(&BraveOperationalProfiling::OnUploadComplete,
+                     base::Unretained(this)));
+}
+
+void BraveOperationalProfiling::OnUploadComplete(
+    scoped_refptr<net::HttpResponseHeaders> headers) {
+  int response_code = -1;
+  if (headers)
+    response_code = headers->response_code();
+  if (response_code == 200) {
+    last_checked_slot_ = current_collected_slot_;
+    SavePrefs();
+  }
+}
+
+std::string BraveOperationalProfiling::BuildPayload() const {
+  base::Value root(base::Value::Type::DICTIONARY);
+
+  root.SetKey("collection_id", base::Value(collection_id_));
+  root.SetKey("platform", base::Value(brave_stats::GetPlatformIdentifier()));
+  root.SetKey("collection_slot", base::Value(current_collected_slot_));
+  root.SetKey("wiki-link", base::Value("https://github.com/brave/brave-browser/"
+                                       "wiki/Anonymous-Operational-Profiles"));
+
+  std::string result;
+  base::JSONWriter::Write(root, &result);
+
+  return result;
+}
+
+int BraveOperationalProfiling::GetCurrentCollectionSlot() const {
+  base::Time::Exploded now;
+  base::Time::Now().LocalExplode(&now);
+
+  return ((now.day_of_month - 1) * 24 * 60 + now.hour * 60 + now.minute) /
```

Comment:
Not for this PR: Change to monotonous slot counter instead of cycling back after one month